chore: output declaration files for neuroglancer packages

chrisj opened the pull request:
esbuild does not support generating declaration files https://github.com/evanw/esbuild/issues/95

chrisj commented on build_tools/build-package.ts:
```diff
@@ -45,13 +64,26 @@ async function buildPackage(options: { inplace?: boolean }) {
 
+  if (declaration) {
+    buildDeclarationFiles(entryPoints, {
+      allowJs: true,
```
need to remove these superfluous arguments

chrisj commented:
I improved the quality of the declarations by using the tsconfig file. The bulk of the change comes from `"moduleResolution": "bundler",`

It does lose a little bit of type information for some reason but it adds a lot more than it removes
```
-    spatiallyIndexedSources: Set<Borrowed<AnnotationGeometryChunkSource>>;
+    spatiallyIndexedSources: Set<AnnotationGeometryChunkSource>;
```

```
-    references: Map<AnnotationId, Borrowed<AnnotationReference>>;
-    localUpdates: Map<AnnotationId, LocalUpdateUndoState>;
+    references: Map<string, AnnotationReference>;
+    localUpdates: Map<string, LocalUpdateUndoState>;
```

jbms commented:
Thanks --- we could probably just emit the declarations all the time --- I don't think there is a time when we don't want the declarations.

jbms commented:
Having the type aliases resolved in some cases is fine, I think, and I agree that some annotations are better than none.

When I tried to generate declarations previously I ran into some errors due to some exported things having types depending on non-exported types, I think.  But I guess it works now for some reason.

chrisj commented:
Removed the flag making it optional. Added that in case the performance benefit of leaving it off was useful.

I agree imperfect types are far better than none. I was fortunate that this solution https://github.com/Microsoft/TypeScript/wiki/Using-the-Compiler-API#getting-the-dts-from-a-javascript-file was very easy to drop in

chrisj commented:
I just realized `const host = ts.createCompilerHost(options);` is probably unnecessary but it does change the declaration files. Removes ".ts" from the imports
```
-    get changed(): import("../util/signal.ts").Signal<() => void>;
+    get changed(): import("../util/signal").Signal<() => void>;
```
Checking to make sure that isn't a problem.

chrisj commented:
`check(new Uint64(1423829346, 3183191017), 28);` in `toString parseString round trip`is a real error. I'm trying to debug it

`check(new Uint64(0, 79741775), 29);` also fails, and it seems to fail with any low value.

79741775 seems to be the smallest high that causes this test to fail with base 29.
